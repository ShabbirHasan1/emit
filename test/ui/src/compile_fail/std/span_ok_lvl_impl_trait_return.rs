fn main() {

}

#[emit::span(ok_lvl: "info", err_lvl: "warn", "try_work")]
fn try_work() -> Result<(), impl std::error::Error> {
    Ok::<(), std::io::Error>(())
}
