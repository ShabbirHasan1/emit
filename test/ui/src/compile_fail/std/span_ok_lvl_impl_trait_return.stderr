error[E0562]: `impl Trait` is not allowed in the type of variable bindings
 --> src/compile_fail/std/span_ok_lvl_impl_trait_return.rs:6:29
  |
6 | fn try_work() -> Result<(), impl std::error::Error> {
  |                             ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `impl Trait` is only allowed in arguments and return types of functions and methods
